test-frame-common/src/main/java/io/skodjob/testframe/resources/ResourceManager.java

```diff
@@ -0,0 +1,252 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class ResourceManager {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ResourceManager.class);
+
+    private static ResourceManager instance;
+    private static KubeClient client;
+    private static KubeCmdClient kubeCmdClient;
+
+    private static ThreadLocal<ExtensionContext> testContext = new ThreadLocal<>();
+
+    private static final Map<String, Stack<ResourceItem>> STORED_RESOURCES = new LinkedHashMap<>();
+
+    public static synchronized ResourceManager getInstance() {
+        if (instance == null) {
+            instance = new ResourceManager();
+            client = new KubeClient();
+            if (TestFrameEnv.CLIENT_TYPE.equals(TestFrameConstants.KUBERNETES)) {
+                kubeCmdClient = new Kubectl();
+            } else {
+                kubeCmdClient = new Oc(client.getKubeconfigPath());
+            }
+        }
+        return instance;
+    }
+
+    public static KubeClient getKubeClient() {
+        return client;
+    }
+
+    public static KubeCmdClient getKubeCmdClient() {
+        return kubeCmdClient;
+    }
+
+    private ResourceType<?>[] resourceTypes;
+
+    public static void setTestContext(ExtensionContext context) {
+        testContext.set(context);
+    }
+
+    public static ExtensionContext getTestContext() {
+        return testContext.get();
+    }
+
+    public final void setResourceTypes(ResourceType... types) {
+        this.resourceTypes = types;
+    }
+
+    public final void pushToStack(ResourceItem item) {
+        synchronized (this) {
+            STORED_RESOURCES.computeIfAbsent(getTestContext().getDisplayName(), k -> new Stack<>());
+            STORED_RESOURCES.get(getTestContext().getDisplayName()).push(item);
+        }
+    }
+
+    public final <T extends HasMetadata> void pushToStack(T resource) {
+        synchronized (this) {
+            STORED_RESOURCES.computeIfAbsent(getTestContext().getDisplayName(), k -> new Stack<>());
+            STORED_RESOURCES.get(getTestContext().getDisplayName()).push(
+                    new ResourceItem<T>(
+                            () -> deleteResource(resource),
+                            resource
+                    ));
+        }
+    }
+
+    @SafeVarargs
+    public final <T extends HasMetadata> void createResourceWithoutWait(T... resources) {
+        createResource(false, resources);
+    }
+
+    @SafeVarargs
+    public final <T extends HasMetadata> void createResourceWithWait(T... resources) {
+        createResource(true, resources);
+    }
+
+    @SafeVarargs
+    private <T extends HasMetadata> void createResource(boolean waitReady, T... resources) {
+        for (T resource : resources) {
+            ResourceType<T> type = findResourceType(resource);
+            pushToStack(resource);
+
+            if (resource.getMetadata().getNamespace() == null) {
+                LOGGER.info("Creating {} {}",
```

Comment:
```suggestion
                LOGGER.info("Creating {}/{}",
```

`kind/resource-name`